SFLoggingKit/Classes/SFLogMessageModifier.swift

```diff
@@ -0,0 +1,237 @@
+        return "\(logLevel.symbol) \(message)"
+    }
+
+}
+
+// MARK: - Log Level Name
+
+open class SFLevelNameLogMessageModifier: SFLogMessageModifier {
+
+    // MARK: - SFMessageModifier
+
+    /// Applies a timestamp to the beginning of the message.
+    ///
+    /// - Parameters:
+    ///   - message:  The original message to format.
+    ///   - logLevel: The log level set for the message.
+    ///   - file: The file in which the call happens.
+    ///   - function: The function in which the call happens.
+    ///   - line: The line at which the call happens.
+    ///
+    /// - Returns: A newly formatted message.
+    open func modify(_ message: String, with logLevel: SFLogLevel, _ file: String, _ function: String, _ line: Int) -> String {
+        return "[\(logLevel.description)] \(message)"
+    }
+
+}
+
+// MARK: - Literals
+
+public struct LiteralOption: OptionSet, Equatable, Hashable {
+
+    // MARK: - Log Levels
+
+    /// Creates a new default `.disabled` instance with a bitmask where all bits are equal to 0.
+    public static let disabled: LiteralOption = LiteralOption(rawValue: 0)
+
+    /// Creates a new default `.file` instance with a bitmask of `1`.
+    public static let file: LiteralOption = LiteralOption(rawValue: 1 << 0)
+
+    /// Creates a new default `.function` instance with a bitmask of `1 << 1`.
+    public static let function: LiteralOption = LiteralOption(rawValue: 1 << 1)
+
+    /// Creates a new default `.line` instance with a bitmask of `1 << 2`.
+    public static let line: LiteralOption = LiteralOption(rawValue: 1 << 2)
+
+    /// Creates a new default `.all` instance with a bitmask where all bits equal are equal to `1`.
+    public static let all: LiteralOption = LiteralOption(rawValue: 0b11111111)
+
+    // MARK: - RawRepresentable
+
+    /// Defines the RawValue type as a UInt8 to satisfy the `RawRepresentable` protocol.
+    public typealias RawValue = UInt8
+
+    /// Returns the raw bitmask value of the LogLevel and satisfies the `RawRepresentable` protocol.
+    public let rawValue: RawValue
+
+    /// Creates a literals options instance with the given raw value.
+    ///
+    /// - Parameter rawValue: The raw bitmask value for the log level.
+    ///
+    /// - Returns: A new log level instance.
+    public init(rawValue: RawValue) {
+        self.rawValue = rawValue
+    }
+
+}
+
+//
+
+open class SFLiteralLogMessageModifier: SFLogMessageModifier {
+
+    // MARK: - Properties
+
+    public let options: LiteralOption
+
+    // NOTE: Add option to choose wheather to keep file extension or not.
+    // NOTE: Add option to change format
+
+    // MARK: - Initializers
+
+    /// Creates an `LiteralExpressionModifier` instance with the specified `options`
+    ///
+    /// - Parameter options: The literals that should be added to the message
+    ///
+    /// - Returns: A new `SFLiteralLogMessageModifier` instance.
+    public init(options: LiteralOption = .all) {
+        self.options = options
```

Comment:
This will allow you either all options or just single one. How would you provide list of options, lets say line and file without method, without extending the current code base?

Reply:
LiteralOption is an [optionset](https://nshipster.com/optionset/). So you can used like this: [. file, . function]